# Experimental stage
%% This package is in experimental stage and should be used only for testing/trial purposes %%  
I will keep improving this with time and feedback.  

![Alt text](https://github.com/microsud/microbiomeutilities/blob/Devel/inst/extdata/images/Anonymous-work-in-progress.png)

[image source clker.com](http://www.clker.com/clipart-work-in-progress.html)

# microbiomeutilities
This is mainly a _wrapper tool package_. Apart for some simple scripts, this package has a single function `run_microbiome_pipeline` for carrying out preliminary QC, Alpha Diversity, Ordination and Composition analysis of OTU tables.  
Check the [repository wiki](https://github.com/microsud/microbiomeutilities/wiki)

**NOTE** the aim of this package is not to replce any of the following tools, instead this package is useful for a quick and (not so) dirty analysis of the OTU tables$ generated by tools such as QIIME (the newer QIIME2) (Caporaso, Kuczynski, Stombaugh et al., 2010), Mothur (Schloss, Westcott, Ryabin et al., 2009), DADA2 (Callahan, McMurdie, Rosen et al., 2016).     

## Why do this?  
The entire process of analysing microbiome data is iterative and you have to invest sufficient time for thorough analysis. However, first it is better to have a quick glance at the data. This helps in planning better the individual analysis such as filtering, normalisation, transformation (all three are topics of hot debate and disagreement). The ability to look quickly if the main factor of interst is having an impact on the community diversity, composition, structure can be useful. This will also help in making better decisions for further in-depth analysis.  
Immediate use can be for people who are working with enrichments and wants these basic analysis only.   

## Direction for this package   
Depending on the real world usefulness, practicality and success, we plan to include complete or parts of this package in the Microbiome R package.  
"Leo Lahti, Sudarshan Shetty [et al.](https://github.com/microbiome/microbiome/graphs/contributors) ([Bioconductor, 2017](https://bioconductor.org/packages/devel/bioc/html/microbiome.html)). Tools for microbiome analysis in R.   

The microbiome R package relies on the independently developed   
[phyloseq](https://github.com/joey711/phyloseq) package and data structures for R-based microbiome analysis developed by Paul McMurdie and Susan Holmes.  
[ggplot2](http://ggplot2.org/) H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2009.  
[tidyverse packages](https://www.tidyverse.org/).  



**Microbiome package website with easy tutorials**:  
URL: [http://microbiome.github.com/microbiome](http://microbiome.github.com/microbiome).  

 
 $ _(OTU or ASVs amplicon sequence variants as suggested recently (Callahan, McMurdie & Holmes, 2017))_.   


The RUniverse for microbiome data analytics has several tools some listed below. 

### Tools:
1.	Ampvis	[Tools for visualising amplicon sequencing data](http://madsalbertsen.github.io/ampvis/)  
2.	CCREPE	[Compositionality Corrected by PErmutation and REnormalization](http://bioconductor.org/packages/release/bioc/html/ccrepe.html)  
3.	DADA2	[Divisive Amplicon Denoising Algorithm](https://www.nature.com/nmeth/journal/v13/n7/full/nmeth.3869.html)  
4.	DESeq2	[Differential expression analysis for sequence count data](https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html)  
5.	edgeR	[empirical analysis of DGE in R](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2796818/)  
6.	mare	[Microbiota Analysis in R Easily](https://github.com/katrikorpela/mare)  
7.	Metacoder	[An R package for visualization and manipulation of community taxonomic diversity data](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005404)  
8.	metagenomeSeq	[Differential abundance analysis for microbial marker-gene surveys](http://www.nature.com/nmeth/journal/v10/n12/full/nmeth.2658.html)  
9.	microbiome R package	[Tools for microbiome analysis in R](https://github.com/microbiome/microbiome)  
10.	MINT	[Multivariate INTegrative method](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1553-8)  
11.	mixDIABLO	[Data Integration Analysis for Biomarker discovery using Latent variable approaches for ‘Omics studies](http://mixomics.org/mixdiablo/)  
12.	mixMC	[Multivariate Statistical Framework to Gain Insight into Microbial Communities](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0160169)  
13.	MMinte	[Methodology for the large-scale assessment of microbial metabolic interactions (MMinte) from 16S rDNA data](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1230-3)  
14.	pathostat	[Statistical Microbiome Analysis on metagenomics results from sequencing data samples](https://bioconductor.org/packages/release/bioc/html/PathoStat.html)  
15.	phylofactor	[Phylogenetic factorization of compositional data](https://peerj.com/articles/2969/)  
16.	phylogeo	[Geographic analysis and visualization of microbiome data](https://www.ncbi.nlm.nih.gov/pubmed/25913208)  
17.	qiimer	[R tools compliment qiime](https://github.com/kylebittinger/qiimer)  
18.	RAM	[R for Amplicon-Sequencing-Based Microbial-Ecology](https://cran.r-project.org/web/packages/RAM/RAM.pdf)  
19.	ShinyPhyloseq	[Web-tool with user interface for Phyloseq](http://joey711.github.io/shiny-phyloseq/)  
20.	SigTree	[Identify and Visualize Significantly Responsive Branches in a Phylogenetic Tree](http://www.sciencedirect.com/science/article/pii/S2001037017300132)  
21.	SPIEC-EASI	[Sparse and Compositionally Robust Inference of Microbial Ecological Networks](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004226)  
22.	structSSI	[Simultaneous and Selective Inference for Grouped or Hierarchically Structured Data](https://www.jstatsoft.org/article/view/v059i13)  
23.	Tax4Fun	[Predicting functional profiles from metagenomic 16S rRNA gene data](https://www.ncbi.nlm.nih.gov/pubmed/25957349)  
24.	taxize	[Taxonomic Information from Around the Web](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3901538/)    
25. labdsv	[Ordination and Multivariate Analysis for Ecology](https://cran.r-project.org/web/packages/labdsv/labdsv.pdf)  
26. Vegan	[R package for community ecologists](https://github.com/vegandevs/vegan)  
27.	igraph	[Network Analysis and Visualization in R](http://igraph.org/r/)

### Useful resources are provided by:  
1. [Ben J. Callahan and Colleagues: Bioconductor Workflow for Microbiome Data Analysis: from raw reads to community analyses](https://f1000research.com/articles/5-1492/v2).   
2. [Comeau AM and Colleagues: Microbiome Helper: a Custom and Streamlined Workflow for Microbiome Research](http://msystems.asm.org/content/2/1/e00127-16)  
3.	MicrobiomeHD	[A standardized database of human gut microbiome studies in health and disease *Case-Control*](http://www.biorxiv.org/content/early/2017/05/08/134031)   
4.	Rhea	[A pipeline with modular R scripts](https://peerj.com/articles/2836/)  
5.	Phyloseq	[Import, share, and analyze microbiome census data using R](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217) 

**Note**: 
A good practise is to use Rmarkdown for documenting your results and sharing with your collaborators and supervisors. For an introduction to [RStudio](https://www.youtube.com/watch?v=cWJzjHh_3kk&t=337s) and an 
[RStudio Overview](https://www.youtube.com/watch?v=n3uue28FD0w)  


#### [github](https://github.com/microsud/Tools-Microbiome-Anlaysis), [twitter](https://twitter.com/gutmicrobe), [googlescholar](https://scholar.google.nl/citations?hl=en&user=Vahc6LUAAAAJ&view_op=list_works&sortby=pubdate), [ORCID ID: 0000-0001-7280-9915](http://orcid.org/0000-0001-7280-9915)   

### References:
1. Callahan, B. J., McMurdie, P. J. & Holmes, S. P. (2017). Exact sequence variants should replace operational taxonomic units in marker gene data analysis. bioRxiv, 113597.  
2. Callahan, B. J., McMurdie, P. J., Rosen, M. J., Han, A. W., Johnson, A. J. A. & Holmes, S. P. (2016). DADA2: high-resolution sample inference from Illumina amplicon data. Nature methods 13, 581-583.  
3. Caporaso, J. G., Kuczynski, J., Stombaugh, J., Bittinger, K., Bushman, F. D., Costello, E. K., Fierer, N., Peña, A. G., Goodrich, J. K. & Gordon, J. I. (2010). QIIME allows analysis of high-throughput community sequencing data. Nature methods 7, 335-336.  
4. Schloss, P. D., Westcott, S. L., Ryabin, T., Hall, J. R., Hartmann, M., Hollister, E. B., Lesniewski, R. A., Oakley, B. B., Parks, D. H. & Robinson, C. J. (2009). Introducing mothur: open-source, platform-independent, community-supported software for describing and comparing microbial communities. Applied and environmental microbiology 75, 7537-7541.  
Team, R. C. (2000). R language definition. Vienna, Austria: R foundation for statistical computing.  


