<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>microbiomeutilities example â€¢ microbiomeutilities</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="microbiomeutilities example">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">microbiomeutilities</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/microbiomeutilities.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>microbiomeutilities example</h1>
            
      
      
      <div class="hidden name"><code>microbiomeutilities.Rmd</code></div>

    </div>

    
    
<p><strong>This package is in experimental stage and should be used only for testing/trial purposes. I will keep improving this with time and feedback.</strong></p>
<p>This is mainly a wrapper tool R package. Apart for some simple scripts for data visualization, this package has a single function called <code>microbiome_pipeline</code> for carrying out preliminary QC, Alpha Diversity, Ordination and Composition analysis of OTU tables. The output is a HTML report for convenient investigating of the data.</p>
<div id="install" class="section level2">
<h2 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">install.packages</span>(<span class="st">"devtools"</span>)
devtools::<span class="kw">install_github</span>(<span class="st">"microsud/microbiomeutilities"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(microbiomeutilities)
<span class="co">#&gt; Loading required package: phyloseq</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 3.4.4</span>
<span class="co">#&gt; Loading required package: microbiome</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; microbiome R package (microbiome.github.com)</span>
<span class="co">#&gt;     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Copyright (C) 2011-2018 Leo Lahti et al. &lt;microbiome.github.io&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'microbiome'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     transform</span>
<span class="kw">library</span>(microbiome)
<span class="kw">library</span>(knitr)
<span class="co">#&gt; Warning: package 'knitr' was built under R version 3.4.4</span>
<span class="kw">library</span>(tibble)
<span class="co">#&gt; Warning: package 'tibble' was built under R version 3.4.3</span></code></pre></div>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>A subset of human intestinal biopsy and faecal data from pre-print <a href="https://www.biorxiv.org/content/early/2017/12/06/227272">Shetty SA et al. 2017</a>.</p>
<p>Useful resources:</p>
<p>For more information you can have a look at <a href="http://joey711.github.io/phyloseq/">Phyloseq</a><br>
Tools for microbiome analysis in R. <strong>Microbiome package version 1.1.2.</strong> URL: <a href="http://microbiome.github.com/microbiome" class="uri">http://microbiome.github.com/microbiome</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">data</span>(<span class="st">"biogeogut"</span>)
ps0 &lt;-<span class="st"> </span>biogeogut

<span class="kw">print</span>(ps0)
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 19063 taxa and 14 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 14 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 19063 taxa by 7 taxonomic ranks ]</span></code></pre></div>
</div>
<div id="microbiome-analysis-pipeline" class="section level2">
<h2 class="hasAnchor">
<a href="#microbiome-analysis-pipeline" class="anchor"></a>Microbiome analysis pipeline</h2>
<p>Function microbiome_pipeline generates an HTML report with preliminary QC, Alpha Diversity, Ordination and Composition analysis of OTU tables. This function save all intermediate files incuding figures and phyloseq objects in user specified directory.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/microbiome_pipeline.html">microbiome_pipeline</a></span>(
  <span class="dt">otufile =</span> <span class="st">"my.biom"</span>,
  <span class="dt">mapping =</span> <span class="st">"mymap.csv"</span>,
  <span class="dt">taxonomy =</span> <span class="ot">NULL</span>,
  <span class="dt">treefilename =</span> <span class="st">"myTree.tre"</span>,
  <span class="dt">type =</span> <span class="st">"biom"</span>,
  <span class="dt">work_dir =</span> <span class="st">"F:/path/my/input/filefolder"</span>,
  <span class="dt">out_dir =</span> <span class="st">"F:/path/to/save/my/files/folder"</span>,
  <span class="dt">VariableA =</span> <span class="st">"MC_type1"</span>,
  <span class="dt">VariableB =</span> <span class="st">"Region"</span>,
  <span class="dt">UnConstOrd =</span> <span class="ot">TRUE</span>,
  <span class="dt">heatmap =</span> <span class="ot">TRUE</span>,
  <span class="dt">filterCount =</span> <span class="dv">4</span>,
  <span class="dt">filterPrev =</span> <span class="fl">0.01</span>,
  <span class="dt">col.palette =</span> <span class="st">"Paired"</span>,
  <span class="dt">filterpseq =</span> <span class="ot">TRUE</span>,
  <span class="dt">samsize =</span> <span class="ot">NA</span>,
  <span class="dt">projectname =</span> <span class="st">"Mock"</span>,
  <span class="dt">author =</span> <span class="st">"Sudarshan"</span>
  )</code></pre></div>
</div>
<div id="formatting-the-phyloseq-object" class="section level2">
<h2 class="hasAnchor">
<a href="#formatting-the-phyloseq-object" class="anchor"></a>Formatting the Phyloseq Object</h2>
<p>Most commonly it is observed that the taxonomy file has classification until a given taxonomic level. We will fill the empty cells with the maximum classification available along with the OTU number.</p>
<p>Check the taxonomy in phyloseq object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">kable</span>(<span class="kw">head</span>(<span class="kw">tax_table</span>(ps0)))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="left">Domain</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>541135</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Firmicutes</td>
<td align="left">c__Bacilli</td>
<td align="left">o__Lactobacillales</td>
<td align="left">f__Lactobacillaceae</td>
<td align="left">g__Lactobacillus</td>
<td align="left">s__</td>
</tr>
<tr class="even">
<td>1050608</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Actinobacteria</td>
<td align="left">c__Actinobacteria</td>
<td align="left">o__Actinomycetales</td>
<td align="left">f__Corynebacteriaceae</td>
<td align="left">g__Corynebacterium</td>
<td align="left">s__</td>
</tr>
<tr class="odd">
<td>370251</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Proteobacteria</td>
<td align="left">c__Gammaproteobacteria</td>
<td align="left">o__Oceanospirillales</td>
<td align="left">f__Endozoicimonaceae</td>
<td align="left">g__</td>
<td align="left">s__</td>
</tr>
<tr class="even">
<td>188753</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Firmicutes</td>
<td align="left">c__Clostridia</td>
<td align="left">o__Clostridiales</td>
<td align="left">f__Lachnospiraceae</td>
<td align="left">g__Blautia</td>
<td align="left">s__</td>
</tr>
<tr class="odd">
<td>217472</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Acidobacteria</td>
<td align="left">c__Acidobacteriia</td>
<td align="left">o__Acidobacteriales</td>
<td align="left">f__Koribacteraceae</td>
<td align="left">g__Candidatus Koribacter</td>
<td align="left">s__</td>
</tr>
<tr class="even">
<td>623634</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Acidobacteria</td>
<td align="left">c__Solibacteres</td>
<td align="left">o__Solibacterales</td>
<td align="left">f__Solibacteraceae</td>
<td align="left">g__Candidatus Solibacter</td>
<td align="left">s__</td>
</tr>
</tbody>
</table>
<p>Some have only g__ of s__ information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
ps0.f &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_phyloseq.html">format_phyloseq</a></span>(ps0)
<span class="co">#&gt; This function will change Kingdom to Domain by default in taxonomy</span>
<span class="co">#&gt; Warning: package 'bindrcpp' was built under R version 3.4.4</span>

<span class="kw">kable</span>(<span class="kw">head</span>(<span class="kw">tax_table</span>(ps0.f)))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="left">Domain</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>541135</td>
<td align="left">Bacteria</td>
<td align="left">Firmicutes</td>
<td align="left">Bacilli</td>
<td align="left">Lactobacillales</td>
<td align="left">Lactobacillaceae</td>
<td align="left">Lactobacillus</td>
<td align="left">f__Lactobacillus_541135</td>
</tr>
<tr class="even">
<td>1050608</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">f__Corynebacterium_1050608</td>
</tr>
<tr class="odd">
<td>370251</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Oceanospirillales</td>
<td align="left">Endozoicimonaceae</td>
<td align="left">f__Endozoicimonaceae_370251</td>
<td align="left">f__Endozoicimonaceae_370251</td>
</tr>
<tr class="even">
<td>188753</td>
<td align="left">Bacteria</td>
<td align="left">Firmicutes</td>
<td align="left">Clostridia</td>
<td align="left">Clostridiales</td>
<td align="left">Lachnospiraceae</td>
<td align="left">Blautia</td>
<td align="left">f__Blautia_188753</td>
</tr>
<tr class="odd">
<td>217472</td>
<td align="left">Bacteria</td>
<td align="left">Acidobacteria</td>
<td align="left">Acidobacteriia</td>
<td align="left">Acidobacteriales</td>
<td align="left">Koribacteraceae</td>
<td align="left">Candidatus Koribacter</td>
<td align="left">f__Candidatus Koribacter_217472</td>
</tr>
<tr class="even">
<td>623634</td>
<td align="left">Bacteria</td>
<td align="left">Acidobacteria</td>
<td align="left">Solibacteres</td>
<td align="left">Solibacterales</td>
<td align="left">Solibacteraceae</td>
<td align="left">Candidatus Solibacter</td>
<td align="left">f__Candidatus Solibacter_623634</td>
</tr>
</tbody>
</table>
<p>Now the available taxonomy is added.<br>
There is a second version which will change the names in both otu table and taxonomy table. This can be useful if the analysis has to be done at OTU level. Only ID are less useful.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
ps0.f2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_to_besthit.html">format_to_besthit</a></span>(ps0)

<span class="kw">kable</span>(<span class="kw">head</span>(<span class="kw">tax_table</span>(ps0.f2)))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="left">Domain</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
<th align="left">best_hit</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>OTU-541135:Lactobacillus</td>
<td align="left">Bacteria</td>
<td align="left">Firmicutes</td>
<td align="left">Bacilli</td>
<td align="left">Lactobacillales</td>
<td align="left">Lactobacillaceae</td>
<td align="left">Lactobacillus</td>
<td align="left">g__Lactobacillus</td>
<td align="left">OTU-541135:Lactobacillus</td>
</tr>
<tr class="even">
<td>OTU-1050608:Corynebacterium</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">g__Corynebacterium</td>
<td align="left">OTU-1050608:Corynebacterium</td>
</tr>
<tr class="odd">
<td>OTU-370251:f__Endozoicimonaceae</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Oceanospirillales</td>
<td align="left">Endozoicimonaceae</td>
<td align="left">f__Endozoicimonaceae</td>
<td align="left">f__Endozoicimonaceae</td>
<td align="left">OTU-370251:f__Endozoicimonaceae</td>
</tr>
<tr class="even">
<td>OTU-188753:Blautia</td>
<td align="left">Bacteria</td>
<td align="left">Firmicutes</td>
<td align="left">Clostridia</td>
<td align="left">Clostridiales</td>
<td align="left">Lachnospiraceae</td>
<td align="left">Blautia</td>
<td align="left">g__Blautia</td>
<td align="left">OTU-188753:Blautia</td>
</tr>
<tr class="odd">
<td>OTU-217472:Candidatus Koribacter</td>
<td align="left">Bacteria</td>
<td align="left">Acidobacteria</td>
<td align="left">Acidobacteriia</td>
<td align="left">Acidobacteriales</td>
<td align="left">Koribacteraceae</td>
<td align="left">Candidatus Koribacter</td>
<td align="left">g__Candidatus Koribacter</td>
<td align="left">OTU-217472:Candidatus Koribacter</td>
</tr>
<tr class="even">
<td>OTU-623634:Candidatus Solibacter</td>
<td align="left">Bacteria</td>
<td align="left">Acidobacteria</td>
<td align="left">Solibacteres</td>
<td align="left">Solibacterales</td>
<td align="left">Solibacteraceae</td>
<td align="left">Candidatus Solibacter</td>
<td align="left">g__Candidatus Solibacter</td>
<td align="left">OTU-623634:Candidatus Solibacter</td>
</tr>
</tbody>
</table>
<p>As can be seen, the rownames have the OTUIDs and available toxonomic name(s).</p>
</div>
<div id="summarize-the-percent-taxa-classification-for-phyloseq" class="section level2">
<h2 class="hasAnchor">
<a href="#summarize-the-percent-taxa-classification-for-phyloseq" class="anchor"></a>Summarize the percent taxa classification for phyloseq</h2>
<p>This can be useful to get an overview of taxonomic classifications. Only patterns such as [g__] or similar is expected. [g__<empty>] or similar not considered pease convert for eg. g__unclassified to uniform [g__].</empty></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/percent_classified.html">percent_classified</a></span>(ps0)
<span class="co">#&gt; Only patterns such as [g__] or similar is expected. [g__&lt;empty&gt;] or [g__unclassified] not considered</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;           please convert for eg. g__unclassified to uniform [g__] or NAs</span>
<span class="co">#&gt;   Taxonomic_Levels Percent_Classification</span>
<span class="co">#&gt; 1           Domain              99.9948 %</span>
<span class="co">#&gt; 2           Phylum              99.9895 %</span>
<span class="co">#&gt; 3            Class              99.8059 %</span>
<span class="co">#&gt; 4            Order              98.4682 %</span>
<span class="co">#&gt; 5           Family              87.3367 %</span>
<span class="co">#&gt; 6            Genus              55.6838 %</span>
<span class="co">#&gt; 7          Species              17.9772 %</span>
<span class="co">#&gt; 8        OTUs/ASVs                  19063</span></code></pre></div>
</div>
<div id="distribution-of-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#distribution-of-reads" class="anchor"></a>Distribution of reads</h2>
<p>Useful for QC purposes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_read_distribution.html">plot_read_distribution</a></span>(ps0, <span class="dt">groups=</span><span class="st">"SampleType"</span>, <span class="dt">plot.type=</span> <span class="st">"density"</span>)
<span class="co">#&gt; [1] "Done plotting"</span>
<span class="kw">print</span>(p)</code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="convert-phyloseq-object-to-long-data-format" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-phyloseq-object-to-long-data-format" class="anchor"></a>Convert phyloseq object to long data format</h2>
<p>Useful in the user wants to plot specific features ot imporve base plots.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
pseq_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/phy_to_ldf.html">phy_to_ldf</a></span>(ps0, <span class="dt">transform.counts =</span> <span class="ot">NULL</span>)
<span class="co">#&gt; An additonal column Sam_rep with sample names is created for reference purpose</span>

<span class="kw">kable</span>(<span class="kw">head</span>(pseq_df))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">OTUID</th>
<th align="left">Domain</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
<th align="left">Species</th>
<th align="left">Sam_rep</th>
<th align="right">Abundance</th>
<th align="left">SampleID</th>
<th align="left">Subject</th>
<th align="left">Age</th>
<th align="left">InputFileName</th>
<th align="right">Sex</th>
<th align="left">SampleType</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">541135</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Firmicutes</td>
<td align="left">c__Bacilli</td>
<td align="left">o__Lactobacillales</td>
<td align="left">f__Lactobacillaceae</td>
<td align="left">g__Lactobacillus</td>
<td align="left">s__</td>
<td align="left">cb110</td>
<td align="right">0</td>
<td align="left">cb110</td>
<td align="left">Subject_110</td>
<td align="left">C</td>
<td align="left">cb110.fasta</td>
<td align="right">1</td>
<td align="left">Biopsy</td>
</tr>
<tr class="even">
<td align="left">1050608</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Actinobacteria</td>
<td align="left">c__Actinobacteria</td>
<td align="left">o__Actinomycetales</td>
<td align="left">f__Corynebacteriaceae</td>
<td align="left">g__Corynebacterium</td>
<td align="left">s__</td>
<td align="left">cb110</td>
<td align="right">5</td>
<td align="left">cb110</td>
<td align="left">Subject_110</td>
<td align="left">C</td>
<td align="left">cb110.fasta</td>
<td align="right">1</td>
<td align="left">Biopsy</td>
</tr>
<tr class="odd">
<td align="left">370251</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Proteobacteria</td>
<td align="left">c__Gammaproteobacteria</td>
<td align="left">o__Oceanospirillales</td>
<td align="left">f__Endozoicimonaceae</td>
<td align="left">g__</td>
<td align="left">s__</td>
<td align="left">cb110</td>
<td align="right">29</td>
<td align="left">cb110</td>
<td align="left">Subject_110</td>
<td align="left">C</td>
<td align="left">cb110.fasta</td>
<td align="right">1</td>
<td align="left">Biopsy</td>
</tr>
<tr class="even">
<td align="left">188753</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Firmicutes</td>
<td align="left">c__Clostridia</td>
<td align="left">o__Clostridiales</td>
<td align="left">f__Lachnospiraceae</td>
<td align="left">g__Blautia</td>
<td align="left">s__</td>
<td align="left">cb110</td>
<td align="right">1</td>
<td align="left">cb110</td>
<td align="left">Subject_110</td>
<td align="left">C</td>
<td align="left">cb110.fasta</td>
<td align="right">1</td>
<td align="left">Biopsy</td>
</tr>
<tr class="odd">
<td align="left">217472</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Acidobacteria</td>
<td align="left">c__Acidobacteriia</td>
<td align="left">o__Acidobacteriales</td>
<td align="left">f__Koribacteraceae</td>
<td align="left">g__Candidatus Koribacter</td>
<td align="left">s__</td>
<td align="left">cb110</td>
<td align="right">0</td>
<td align="left">cb110</td>
<td align="left">Subject_110</td>
<td align="left">C</td>
<td align="left">cb110.fasta</td>
<td align="right">1</td>
<td align="left">Biopsy</td>
</tr>
<tr class="even">
<td align="left">623634</td>
<td align="left">k__Bacteria</td>
<td align="left">p__Acidobacteria</td>
<td align="left">c__Solibacteres</td>
<td align="left">o__Solibacterales</td>
<td align="left">f__Solibacteraceae</td>
<td align="left">g__Candidatus Solibacter</td>
<td align="left">s__</td>
<td align="left">cb110</td>
<td align="right">0</td>
<td align="left">cb110</td>
<td align="left">Subject_110</td>
<td align="left">C</td>
<td align="left">cb110.fasta</td>
<td align="right">1</td>
<td align="left">Biopsy</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-alpha-diversities" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-alpha-diversities" class="anchor"></a>Plot alpha diversities</h2>
<p>Utility plot function for diversity measures calcualted by microbiome package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(microbiome)
<span class="kw">data</span>(<span class="st">"biogeogut"</span>)
ps0 &lt;-<span class="st"> </span>biogeogut

p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_alpha_diversities.html">plot_alpha_diversities</a></span>(ps0, 
                            <span class="dt">type =</span> <span class="st">"diversities"</span>, 
                            <span class="dt">index.val =</span> <span class="st">"all"</span>, 
                            <span class="dt">plot.type =</span> <span class="st">"stripchart"</span>, 
                            <span class="dt">variableA =</span> <span class="st">"SampleType"</span>, 
                            <span class="dt">palette =</span> <span class="st">"jco"</span>)
<span class="co">#&gt; Using sam_rep_nw as id variables</span>
<span class="co">#&gt; Using SampleID, Subject, Age, InputFileName, SampleType, sam_rep_nw as id variables</span>

<span class="kw">print</span>(p)</code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-11-1.png" width="576"></p>
</div>
<div id="plot-ordination-and-core" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-ordination-and-core" class="anchor"></a>Plot ordination and core</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phyloseq)
<span class="kw">library</span>(microbiome)
<span class="kw">data</span>(<span class="st">"atlas1006"</span>)
pseq &lt;-<span class="st"> </span>atlas1006

pseq.rel &lt;-<span class="st"> </span>microbiome::<span class="kw">transform</span>(pseq, <span class="st">"compositional"</span>)

pseq.core &lt;-<span class="st"> </span><span class="kw">subset_samples</span>(pseq.rel, gender ==<span class="st"> "female"</span> &amp;<span class="st"> </span>DNA_extraction_method ==<span class="st"> "r"</span> &amp;<span class="st"> </span>bmi_group ==<span class="st"> "lean"</span>)

prev.thres &lt;-<span class="st"> </span><span class="kw">seq</span>(.<span class="dv">05</span>, <span class="dv">1</span>, .<span class="dv">05</span>)
det.thres &lt;-<span class="st"> </span><span class="dv">10</span>^<span class="kw">seq</span>(<span class="kw">log10</span>(<span class="fl">1e-2</span>), <span class="kw">log10</span>(.<span class="dv">2</span>), <span class="dt">length =</span> <span class="dv">10</span>)

<span class="co"># Currently only NMDS is supported</span>
ord.bray &lt;-<span class="st"> </span><span class="kw">ordinate</span>(pseq.core, <span class="st">"NMDS"</span>, <span class="st">"bray"</span>)
<span class="co">#&gt; Run 0 stress 0.2101503 </span>
<span class="co">#&gt; Run 1 stress 0.210127 </span>
<span class="co">#&gt; ... New best solution</span>
<span class="co">#&gt; ... Procrustes: rmse 0.002221749  max resid 0.01918572 </span>
<span class="co">#&gt; Run 2 stress 0.2432069 </span>
<span class="co">#&gt; Run 3 stress 0.2101322 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.0008194299  max resid 0.005902171 </span>
<span class="co">#&gt; ... Similar to previous best</span>
<span class="co">#&gt; Run 4 stress 0.21012 </span>
<span class="co">#&gt; ... New best solution</span>
<span class="co">#&gt; ... Procrustes: rmse 0.001254357  max resid 0.010399 </span>
<span class="co">#&gt; Run 5 stress 0.2472289 </span>
<span class="co">#&gt; Run 6 stress 0.2384332 </span>
<span class="co">#&gt; Run 7 stress 0.2101189 </span>
<span class="co">#&gt; ... New best solution</span>
<span class="co">#&gt; ... Procrustes: rmse 0.0006416618  max resid 0.004980311 </span>
<span class="co">#&gt; ... Similar to previous best</span>
<span class="co">#&gt; Run 8 stress 0.2373057 </span>
<span class="co">#&gt; Run 9 stress 0.2101182 </span>
<span class="co">#&gt; ... New best solution</span>
<span class="co">#&gt; ... Procrustes: rmse 0.0002021048  max resid 0.001467387 </span>
<span class="co">#&gt; ... Similar to previous best</span>
<span class="co">#&gt; Run 10 stress 0.2416971 </span>
<span class="co">#&gt; Run 11 stress 0.2101199 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.0002693704  max resid 0.001604083 </span>
<span class="co">#&gt; ... Similar to previous best</span>
<span class="co">#&gt; Run 12 stress 0.2471126 </span>
<span class="co">#&gt; Run 13 stress 0.2101474 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.001941356  max resid 0.01923798 </span>
<span class="co">#&gt; Run 14 stress 0.246152 </span>
<span class="co">#&gt; Run 15 stress 0.2101305 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.001133971  max resid 0.0104276 </span>
<span class="co">#&gt; Run 16 stress 0.2101269 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.001149552  max resid 0.01029439 </span>
<span class="co">#&gt; Run 17 stress 0.2101174 </span>
<span class="co">#&gt; ... New best solution</span>
<span class="co">#&gt; ... Procrustes: rmse 0.0003128505  max resid 0.001855264 </span>
<span class="co">#&gt; ... Similar to previous best</span>
<span class="co">#&gt; Run 18 stress 0.2101195 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.0004448604  max resid 0.002765298 </span>
<span class="co">#&gt; ... Similar to previous best</span>
<span class="co">#&gt; Run 19 stress 0.2205393 </span>
<span class="co">#&gt; Run 20 stress 0.2101623 </span>
<span class="co">#&gt; ... Procrustes: rmse 0.002392057  max resid 0.02154235 </span>
<span class="co">#&gt; *** Solution reached</span>

px &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_ordiplot_core.html">plot_ordiplot_core</a></span>(pseq.core, <span class="dt">ordiObject =</span> ord.bray, <span class="dt">coreplot =</span> <span class="ot">TRUE</span>, <span class="dt">prevalences =</span> prev.thres, <span class="dt">detections =</span> det.thres, <span class="dt">color.opt =</span> <span class="st">"diversity"</span>, <span class="dt">shape =</span> <span class="ot">NULL</span>)</code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-12-1.png" width="1152"></p>
</div>
<div id="visualize-abundance-of-select-taxa" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-abundance-of-select-taxa" class="anchor"></a>Visualize abundance of select taxa</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
ps.f2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/format_to_besthit.html">format_to_besthit</a></span>(ps0)
psf2.rel &lt;-<span class="st"> </span>microbiome::<span class="kw">transform</span>(ps.f2, <span class="st">"compositional"</span>)
otu &lt;-<span class="st"> </span><span class="kw">abundances</span>(psf2.rel)
meta &lt;-<span class="st"> </span><span class="kw">meta</span>(psf2.rel)

<span class="kw">library</span>(vegan)
<span class="co">#&gt; Loading required package: permute</span>
<span class="co">#&gt; Loading required package: lattice</span>
<span class="co">#&gt; This is vegan 2.4-4</span>
permanova &lt;-<span class="st"> </span><span class="kw">adonis</span>(<span class="kw">t</span>(otu) ~<span class="st"> </span>SampleType,
               <span class="dt">data =</span> meta, <span class="dt">permutations=</span><span class="dv">99</span>, <span class="dt">method =</span> <span class="st">"bray"</span>)

<span class="co"># P-value</span>
<span class="kw">print</span>(<span class="kw">as.data.frame</span>(permanova$aov.tab)[<span class="st">"group"</span>, <span class="st">"Pr(&gt;F)"</span>])
<span class="co">#&gt; [1] NA</span>

coef &lt;-<span class="st"> </span><span class="kw">coefficients</span>(permanova)[<span class="st">"SampleType1"</span>,]
top.coef &lt;-<span class="st"> </span>coef[<span class="kw">rev</span>(<span class="kw">order</span>(<span class="kw">abs</span>(coef)))[<span class="dv">1</span>:<span class="dv">6</span>]]

top.coef.df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(top.coef)

my_taxa &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rownames</span>(top.coef.df))

p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_select_taxa.html">plot_select_taxa</a></span>(psf2.rel, my_taxa, <span class="st">"SampleType"</span>, <span class="st">"Paired"</span>, <span class="dt">plot.type =</span> <span class="st">"boxplot"</span>)
<span class="co">#&gt; An additonal column Sam_rep with sample names is created for reference purpose</span>
p</code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
</div>
<div id="plot-taxa-boxplot" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-taxa-boxplot" class="anchor"></a>Plot taxa boxplot</h2>
<p>Plot relative abundance of top taxa specified by user.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
pn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_taxa_boxplot.html">plot_taxa_boxplot</a></span>(ps0,
                        <span class="dt">taxonomic.level =</span> <span class="st">"Phylum"</span>,
                        <span class="dt">top.otu =</span> <span class="dv">5</span>, <span class="dt">VariableA =</span> <span class="st">"SampleType"</span>,
                        <span class="dt">title =</span> <span class="st">"Relative abudance plot"</span>, <span class="dt">color =</span> <span class="st">"Set2"</span>)
<span class="co">#&gt; The phy_tree slot is empty, easy to make the plot</span>

<span class="kw">print</span>(pn)</code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-14-1.png" width="576"></p>
</div>
<div id="plot-taxa-barplot" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-taxa-barplot" class="anchor"></a>Plot taxa barplot</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(RColorBrewer)
<span class="kw">library</span>(dplyr)
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 3.4.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>

ps.cor &lt;-<span class="st"> </span><span class="kw">core</span>(ps0, <span class="dt">detection =</span> <span class="dv">10</span>, <span class="dt">prevalence =</span> <span class="fl">0.5</span>)
p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_taxa_composition.html">plot_taxa_composition</a></span>(ps.cor, <span class="dt">taxonomic.level =</span> <span class="st">"Phylum"</span>, <span class="dt">transform =</span> <span class="st">"compositional"</span>, <span class="dt">average_by =</span> <span class="ot">NULL</span>)
p</code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-15-1.png" width="576"></p>
</div>
<div id="heatmap-using-phyloseq-and-pheatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmap-using-phyloseq-and-pheatmap" class="anchor"></a>Heatmap using phyloseq and pheatmap</h2>
<p>Useful for visualisng differences in top otus between sample groups.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
heat.sample &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plot_taxa_heatmap.html">plot_taxa_heatmap</a></span>(ps0, <span class="dt">subset.top =</span> <span class="dv">20</span>,
    <span class="dt">VariableA =</span> <span class="st">"SampleType"</span>,
    <span class="dt">heatcolors =</span><span class="kw">brewer.pal</span>(<span class="dv">100</span>, <span class="st">"Blues"</span>),
    <span class="dt">transformation =</span> <span class="st">"log10"</span>)
<span class="co">#&gt; Top 20 OTUs selected</span>
<span class="co">#&gt; log10, if zeros in data then log10(1+x) will be used</span>
<span class="co">#&gt; First top taxa were selected and </span>
<span class="co">#&gt; then abundances tranformed to log10(1+X)</span>
<span class="co">#&gt; Warning in transform(phyobj1, "log10"): OTU table contains zeroes. Using</span>
<span class="co">#&gt; log10(1 + x) transform.</span>
<span class="co">#&gt; Warning in brewer.pal(100, "Blues"): n too large, allowed maximum for palette Blues is 9</span>
<span class="co">#&gt; Returning the palette you asked for with that many colors</span></code></pre></div>
<p><img src="microbiomeutilities_files/figure-html/unnamed-chunk-16-1.png" width="768"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.4.1 (2017-06-30)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 7 x64 (build 7601) Service Pack 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=Dutch_Netherlands.1252  LC_CTYPE=Dutch_Netherlands.1252   </span>
<span class="co">#&gt; [3] LC_MONETARY=Dutch_Netherlands.1252 LC_NUMERIC=C                      </span>
<span class="co">#&gt; [5] LC_TIME=Dutch_Netherlands.1252    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] dplyr_0.7.4                RColorBrewer_1.1-2        </span>
<span class="co">#&gt;  [3] vegan_2.4-4                lattice_0.20-35           </span>
<span class="co">#&gt;  [5] permute_0.9-4              bindrcpp_0.2.2            </span>
<span class="co">#&gt;  [7] tibble_1.4.2               knitr_1.20                </span>
<span class="co">#&gt;  [9] microbiomeutilities_0.99.0 microbiome_1.1.10012      </span>
<span class="co">#&gt; [11] ggplot2_2.2.1              phyloseq_1.23.1           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] viridis_0.5.1       Biobase_2.36.2      tidyr_0.8.0        </span>
<span class="co">#&gt;  [4] viridisLite_0.3.0   jsonlite_1.5        splines_3.4.1      </span>
<span class="co">#&gt;  [7] foreach_1.4.4       assertthat_0.2.0    highr_0.6          </span>
<span class="co">#&gt; [10] stats4_3.4.1        yaml_2.1.18         ggrepel_0.7.0      </span>
<span class="co">#&gt; [13] pillar_1.2.1        backports_1.1.2     glue_1.2.0         </span>
<span class="co">#&gt; [16] digest_0.6.15       XVector_0.16.0      colorspace_1.3-2   </span>
<span class="co">#&gt; [19] cowplot_0.9.2       htmltools_0.3.6     Matrix_1.2-14      </span>
<span class="co">#&gt; [22] plyr_1.8.4          pkgconfig_2.0.1     pheatmap_1.0.8     </span>
<span class="co">#&gt; [25] zlibbioc_1.22.0     purrr_0.2.4         scales_0.5.0       </span>
<span class="co">#&gt; [28] Rtsne_0.13          mgcv_1.8-23         IRanges_2.10.5     </span>
<span class="co">#&gt; [31] ggpubr_0.1.6        BiocGenerics_0.22.1 lazyeval_0.2.1     </span>
<span class="co">#&gt; [34] survival_2.41-3     magrittr_1.5        crayon_1.3.4       </span>
<span class="co">#&gt; [37] memoise_1.1.0       evaluate_0.10.1     fs_1.2.2           </span>
<span class="co">#&gt; [40] nlme_3.1-137        MASS_7.3-49         xml2_1.2.0         </span>
<span class="co">#&gt; [43] tools_3.4.1         data.table_1.10.4-3 stringr_1.3.0      </span>
<span class="co">#&gt; [46] S4Vectors_0.14.7    munsell_0.4.3       ggsci_2.8          </span>
<span class="co">#&gt; [49] cluster_2.0.7-1     Biostrings_2.44.2   ade4_1.7-11        </span>
<span class="co">#&gt; [52] compiler_3.4.1      pkgdown_0.1.0.9000  rlang_0.2.0.9001   </span>
<span class="co">#&gt; [55] rhdf5_2.20.0        grid_3.4.1          iterators_1.0.9    </span>
<span class="co">#&gt; [58] biomformat_1.4.0    rstudioapi_0.7      igraph_1.2.1       </span>
<span class="co">#&gt; [61] labeling_0.3        rmarkdown_1.9       gtable_0.2.0       </span>
<span class="co">#&gt; [64] codetools_0.2-15    multtest_2.32.0     roxygen2_6.0.1     </span>
<span class="co">#&gt; [67] reshape2_1.4.3      R6_2.2.2            gridExtra_2.3      </span>
<span class="co">#&gt; [70] bindr_0.1.1         commonmark_1.4      rprojroot_1.3-2    </span>
<span class="co">#&gt; [73] desc_1.1.1          ape_5.1             stringi_1.1.7      </span>
<span class="co">#&gt; [76] parallel_3.4.1      Rcpp_0.12.16        tidyselect_0.2.4</span></code></pre></div>
</div>
<div id="package-requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#package-requirements" class="anchor"></a>Package requirements</h2>
<ul>
<li>Depends:</li>
<li><a href="https://www.r-project.org/">R (&gt;= 3.4.1)</a></li>
<li><a href="https://joey711.github.io/phyloseq/index.html">phyloseq</a></li>
<li><a href="http://ggplot2.org/">ggplot2</a></li>
<li><p><a href="https://bioconductor.org/packages/devel/bioc/html/microbiome.html">microbiome</a></p></li>
<li>Imports:</li>
<li><a href="https://cran.r-project.org/web/packages/plyr/index.html">plyr</a></li>
<li><a href="https://cran.rstudio.com/web/packages/reshape2/index.html">reshape2</a></li>
<li><a href="https://stat.ethz.ch/R-manual/R-devel/library/stats/html/00Index.html">stats</a></li>
<li><a href="http://tidyr.tidyverse.org/">tidyr</a></li>
<li><a href="https://stat.ethz.ch/R-manual/R-devel/library/utils/html/00Index.html">utils</a></li>
<li><a href="https://cran.r-project.org/web/packages/data.table/index.html">data.table</a></li>
<li><a href="https://cran.r-project.org/web/packages/vegan/index.html">vegan</a></li>
<li><a href="https://cran.r-project.org/web/packages/ggpubr/index.html">ggpubr</a></li>
<li><a href="https://cran.r-project.org/web/packages/RColorBrewer/index.html">RColorBrewer</a></li>
<li><a href="https://cran.r-project.org/web/packages/viridis/index.html">viridis</a></li>
<li><a href="https://cran.r-project.org/web/packages/pheatmap/index.html">pheatmap</a></li>
<li><p><a href="https://cran.r-project.org/web/packages/ggrepel/index.html">ggrepel</a></p></li>
<li>Suggests:</li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/BiocGenerics.html">BiocGenerics</a></li>
<li><a href="https://cran.r-project.org/web/packages/knitr/index.html">knitr</a></li>
<li><a href="http://rmarkdown.rstudio.com/">rmarkdown</a></li>
<li><a href="https://cran.r-project.org/web/packages/ape/index.html">ape</a></li>
<li><p><a href="https://academic.oup.com/bioinformatics/article/26/11/1463/203321">Picante</a></p></li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#install">Install</a></li>
      <li><a href="#example-data">Example data</a></li>
      <li><a href="#microbiome-analysis-pipeline">Microbiome analysis pipeline</a></li>
      <li><a href="#formatting-the-phyloseq-object">Formatting the Phyloseq Object</a></li>
      <li><a href="#summarize-the-percent-taxa-classification-for-phyloseq">Summarize the percent taxa classification for phyloseq</a></li>
      <li><a href="#distribution-of-reads">Distribution of reads</a></li>
      <li><a href="#convert-phyloseq-object-to-long-data-format">Convert phyloseq object to long data format</a></li>
      <li><a href="#plot-alpha-diversities">Plot alpha diversities</a></li>
      <li><a href="#plot-ordination-and-core">Plot ordination and core</a></li>
      <li><a href="#visualize-abundance-of-select-taxa">Visualize abundance of select taxa</a></li>
      <li><a href="#plot-taxa-boxplot">Plot taxa boxplot</a></li>
      <li><a href="#plot-taxa-barplot">Plot taxa barplot</a></li>
      <li><a href="#heatmap-using-phyloseq-and-pheatmap">Heatmap using phyloseq and pheatmap</a></li>
      <li><a href="#package-requirements">Package requirements</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Sudarshan Shetty, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
